rplidar.go

```diff
@@ -348,44 +365,76 @@ func (d *Device) scan(options lidar.ScanOptions) (lidar.Measurements, error) {
 
 			nodeAngle := (float64(node.GetAngle_z_q14()) * 90 / (1 << 14))
 			nodeDistance := float64(node.GetDist_mm_q2()) / 4
-			measurements = append(measurements, lidar.NewMeasurement(nodeAngle, nodeDistance/1000))
+
+			err := pc.Set(pointFrom(rdkUtils.DegToRad(nodeAngle), rdkUtils.DegToRad(0), float64(nodeDistance)/1000, 255))
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	if len(measurements) == 0 {
+	if pc.Size() == 0 {
 		return nil, nil
 	}
-	if options.NoFilter {
-		return measurements, nil
-	}
-	sort.Stable(measurements)
-	filteredMeasurements := make(lidar.Measurements, 0, len(measurements))
-
-	minAngleDiff, maxDistDiff := d.filterParams()
-	prev := measurements[0]
-	detectedRay := false
-	for mIdx := 1; mIdx < len(measurements); mIdx++ {
-		curr := measurements[mIdx]
-		currAngle := curr.AngleDeg()
-		prevAngle := prev.AngleDeg()
-		currDist := curr.Distance()
-		prevDist := prev.Distance()
-		if math.Abs(currAngle-prevAngle) < minAngleDiff {
-			if math.Abs(currDist-prevDist) > maxDistDiff {
-				detectedRay = true
-				continue
-			}
-		}
-		prev = curr
-		if !detectedRay {
-			filteredMeasurements = append(filteredMeasurements, curr)
+	return pc, nil
+}
+
+func (d *Device) savePCDFile(timeStamp time.Time, pc pointcloud.PointCloud) error {
+	f, err := os.Create("data/rplidar_data_" + timeStamp.Format("2006-01-02T15_04_05.0000") + ".pcd")
+	if err != nil {
+		return err
+	}
+
+	w := bufio.NewWriter(f)
+	if err = pc.ToPCD(w); err != nil {
+		return err
+	}
+	if err = w.Flush(); err != nil {
+		return err
+	}
+	if err = f.Close(); err != nil {
+		return err
+	}
+	return nil
```

Comment:
[nit] You can slightly simplify these last few lines by directly returning f.Close() instead of checking if it's an error.

`return f.Close()`

Reply:
But otherwise, good catch to flush the buffer before closing the backing file too!